client/src/pages/Profile.tsx

```diff
@@ -0,0 +1,204 @@
+import { useState, useEffect } from "react";
+import {
+  Box,
+  Flex,
+  Text,
+  Avatar,
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  ModalFooter,
+  Input,
+  Textarea,
+  useToast,
+  Heading,
+} from "@chakra-ui/react";
+import { useSelector } from "react-redux";
+import { userSelector } from "redux/user";
+
+/**
+ * The user profile component.
+ */
+const Profile = () => {
+  // The URL of the API server.
+  const API_URL = `${import.meta.env.VITE_API_SERVER_URL}/api/v1`;
+  const AuthToken = localStorage.getItem("BEARER_TOKEN") || "";
+
+  // The user object from the Redux store.
+  const { user } = useSelector(userSelector);
+
+  // Whether or not the user is in editing mode.
+  const [isEditing, setIsEditing] = useState(false);
+
+  // The display name, email, avatar URL, bio, and body of the user.
+  const [displayName, setDisplayName] = useState("");
+  const [email, setEmail] = useState("");
+  const [avatarUrl, setAvatarUrl] = useState("");
+  const [bio, setBio] = useState("");
+  const [body, setBody] = useState("");
```

Comment:
You need to separate the edit state (form state) and the actual state. Create a `ProfileForm `component and move the form logic there (see `PostForm.tsx`) and use this for fetching and rendering data (see `pages/Property.tsx`)